### PR TITLE
doc: add run instructions for quartz/lassen

### PR DESCRIPTION
**Pull request by matthiasdiener**
This is how it looks like rendered: https://mirgecom.readthedocs.io/en/doc_run/running.html#running-on-specific-systems

### CODE REVIEW

File: doc/running.rst [hunk 1]
```diff
@@ -42,7 +42,8 @@ In order to avoid the slowdowns and warnings, users can direct the packages to c
 cache files in directories that are private to each rank by using the ``XDG_CACHE_HOME``
 environment variable, such as in the following example::
 
-   $ srun -n 512 bash -c 'XDG_CACHE_HOME=/p/lscratchh/diener3/xdg-scratch$SLURM_PROCID python examples/wave-eager-mpi.py'
+   $ export XDG_CACHE_HOME="/tmp/$USER/xdg-scratch"
+   $ srun -n 512 python examples/wave-eager-mpi.py'
```
Comment (inducer) [inline doc/running.rst:46]: Will this automatically propagate env vars?

File: doc/running.rst [hunk 2]
```diff
@@ -59,3 +60,62 @@ In addition to using this array context, you also need to enable profiling in th
...
+In general, we recommend running mirgecom with 1 MPI rank (= python process) per node.
+Kernel execution will be parallelized automatically through pocl
+(either on CPU or GPU, depending on the options you selected and what is available
+on the system).
+
+
+Quartz
+******
+
+On the Quartz machine, running mirgecom should be straightforward. An example batch script for the slurm batch system is given below:
```
Comment (inducer) [inline doc/running.rst:80]: Link to machine docs

File: doc/running.rst [hunk 3]
```diff
@@ -59,3 +60,62 @@ In addition to using this array context, you also need to enable profiling in th
...
+
+   srun -n $nproc python examples/wave-eager-mpi.py
+
+Run this with ``sbatch <script.sh>``.
+
+
+Lassen
+******
+
+On Lassen, we recommend running 1 MPI rank per GPU on each node. Care must be
```
Comment (inducer) [inline doc/running.rst:101]: Link to machine docs

**Review by inducer — COMMENTED** [review]
Thanks for taking this on. This is super important.

File: doc/running.rst (hunk 1 above)
Comment (matthiasdiener) [reply doc/running.rst:46]: In my tests, yes.

File: doc/running.rst (hunk 2 above)
Comment (matthiasdiener) [reply doc/running.rst:80]: Done.

File: doc/running.rst (hunk 3 above)
Comment (matthiasdiener) [reply doc/running.rst:105]: Done

**Comment by matthiasdiener** [comment]
Needs review.

**Review by majosm — CHANGES_REQUESTED** [review]

File: doc/running.rst [hunk 4]
```diff
@@ -59,3 +60,74 @@ In addition to using this array context, you also need to enable profiling in th
...
+
+   #!/bin/bash
+   #SBATCH -t 00:30:00 # walltime
+   #SBATCH -p pbatch
+
+   nnodes=$SLURM_JOB_NUM_NODES
+   nproc=$nnodes # 1 rank per node
+
+   export XDG_CACHE_HOME="/tmp/$USER/xdg-scratch" # See above on why this is important
+
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply doc/running.rst:95]: AFAIK, the conda env remains intact if it is loaded on the head node.

File: doc/running.rst [hunk 5]
```diff
@@ -42,7 +42,8 @@ In order to avoid the slowdowns and warnings, users can direct the packages to c
 cache files in directories that are private to each rank by using the ``XDG_CACHE_HOME``
```
Comment (majosm) [inline doc/running.rst:42]:
```suggestion
cache files in directories that are private to each node by using the ``XDG_CACHE_HOME``
```
This suggestion disappeared from my original review for some reason... Anyway, is it per-rank or per-node? per-node, right?

Comment (matthiasdiener) [reply doc/running.rst:42]: Sorry, did I misclick sth? In the way it is currently implemented, it is per-node.

Comment (majosm) [reply doc/running.rst]: 🤷‍♂️ Github ate it, I guess.

Comment (matthiasdiener) [reply doc/running.rst]: "The dog ate my pull requests" 🤣

**Review by majosm — APPROVED** [review]
Looks good on my end.

**Review by inducer — APPROVED** [review]
Thanks! LGTM

**Pull request merged**